src/main/java/com/example/demo/controllers/AccountController.java

```diff
@@ -70,4 +78,40 @@ public UserModel forgotPassword(@RequestBody UserModel data) throws AuctionAppEx
 	public UserModel newPassword(@RequestBody UserModel data) throws AuctionAppException {
 		return accountService.newPassword(data);
 	}
+
+	@ApiOperation(value = "Updates account data", notes = "Only authenticated users")
+	@PutMapping("api/account")
+	public UserModel updateData(@RequestBody UserModel data) throws AuctionAppException{
+		User principal=null;
+		try {
+			principal = (User)SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+		}catch(ClassCastException ex) {
+			throw new UnauthenticatedException();
+		}
```

Comment:
since this code is repeated a lot, how about we catch this ClassCastException in filter and in these methods to be sure that we can just do `SecurityContextHolder.getContext().getAuthentication().getPrincipal()`

Reply:
Need help...

Reply:
We can leave this refactoring for later, it is nice to have, but not a big deal